packages/dds/tree/docs/data-model/README.md

```diff
@@ -0,0 +1,166 @@
+
+From the perspective of the broader Fluid system, node values are [*serializable*](https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/datastore-definitions/src/serializable.ts)
+and are traversed by Fluid services such as garbage collection.
+
+### Field
+
+Each tree node has a set of zero or more fields.
+Fields are used to model map-like composite type, where each  ***field*** represents one relationship between a parent node and an ordered ***sequence*** of one or more children.
+
+<figure align="center">
+  <img src="./assets/children.drawio.svg" alt=""/>
+  <figcaption>Figure: A parent node with two fields</figcaption>
+</figure>
+
+The fields of a node are distinguished by a ***key***.
+Fields of the same node are distinguished by a field key.
+From the data model's perspective, field keys are opaque.
+
+### Sequences
+
+In the SharedTree data model abstraction, there is no distinction between a field that always contain a single child and a field that can contain multiple children.
+Instead, SharedTree treats all fields as ordered collections and uses schema to restrict which fields are optional, must contain a single value, or may contain multiple values.
+
+When mapping the SharedTree data model to conventional programming languages, it is sometimes helpful to think of the sequence as being distinct from the field, as if each field points to an ***implicit sequence*** object.
+
+However, it is important to remember that sequences are not tree nodes.
+This means that sequences are not directly addressable.
+The combination of the field + sequences is implicitly created when the first item is inserted, implicitly deleted when the last item is removed, and may only be referenced indirectly via the combination of parent node + field key.
+
+### Special Fields
+
+This section covers fields that receive special treatment in the SharedTree data model.
+These fields are special because:
+
+- They have well-known keys
+- They are universally available on all nodes (regardless of schema.)
+- They can not be targeted by normal tree operations.
+
+#### Type
+
+The SharedTree data model supports nominal typing via a special *type* field.
+The value of the *type* field is the unique identifier of the corresponding schema type.
+
+In the data model abstraction, the value of the type field is opaque.
+However, there is a set of well-known types (*boolean*, *number*, etc.) that are transparent to the underlying implementation.
+
+#### Id
+
+The SharedTree data model supports durable references to tree nodes via a special *id* field.
+The *id* field is used by SharedTree to provide a bidirectional *id* ⟷ *node* index that provides efficient lookup of nodes by id.
+This is the underlying building block for features like "share link" URLs and graph-like references within the tree.
+
+## JSON Comparison
+
+The below diagram highlights the differences between the JSON data model and the SharedTree data model using the following snippet:
```

Comment:
That description of the diagram below had me expecting the `dashStyle` field would have had the two children directly under it in the SharedTree diragram (as opposed to having an intermediate node). As is, I think the diagram illustrates how the JSON data model may be mapped onto the ShareTree data model, which is a different thing. I suspect we either want to show both (separately) or only show the mapping, in which case this sentence could be ammended to better reflect that.

Reply:
I added a new section on implicit sequences and explicit arrays above.  I also added comments to the JSON to indicate which one the example is using for which fields.